rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,403 @@
+---
+Start Date: 11-02-2022
+RFC PR: (leave this empty if no PR yet)
+Blade Issue: (leave this empty if no issue yet)
+---
+
+# Responsive and Adaptive Layout Strategy <!-- omit in toc -->
+
+### Table Of Contents <!-- omit in toc -->
+
+- [Summary](#summary)
+- [Basic Example](#basic-example)
+    - [Responsive Modal](#responsive-modal)
+    - [Adaptive Search Field](#adaptive-search-field)
+- [Motivation](#motivation)
+    - [Why are we doing this?](#why-are-we-doing-this)
+    - [What use cases does it support?](#what-use-cases-does-it-support)
+    - [What is the expected outcome?](#what-is-the-expected-outcome)
+- [Detailed Design](#detailed-design)
+  - [Responsive Design](#responsive-design)
+    - [Pros and Cons of Responsive Design](#pros-and-cons-of-responsive-design)
+  - [Adaptive Design](#adaptive-design)
+    - [Pros and Cons of Adaptive Design](#pros-and-cons-of-adaptive-design)
+  - [Approaches taken by well-known Design Systems](#approaches-taken-by-well-known-design-systems)
+  - [What would work for us?](#what-would-work-for-us)
+    - [1. Components that can handle Responsiveness Internally](#1-components-that-can-handle-responsiveness-internally)
+    - [2. Components that can give some flexibility to handle Responsiveness to consumers](#2-components-that-can-give-some-flexibility-to-handle-responsiveness-to-consumers)
+    - [3. Components for Adaptive layout(rendering components conditionally)](#3-components-for-adaptive-layoutrendering-components-conditionally)
+  - [How will we actually implement the approaches discussed above?](#how-will-we-actually-implement-the-approaches-discussed-above)
+    - [On Blade Code Side](#on-blade-code-side)
+    - [On Consumer Code Side](#on-consumer-code-side)
+    - [On Blade Design Side](#on-blade-design-side)
+    - [On Consumer Design Side](#on-consumer-design-side)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+# Summary
+
+To render content on different device screen resolutions there are different strategies. Modern web design gives us three options to use: Responsive, Adaptive, and Standalone Design, although standalone has fallen into disuse because of the overhead it comes with to maintain multiple codebase and their development lifecycle hence, we would just be focussing on Adaptive and Responsive.
+
+1. Responsive - Make the content fluid so it'll render based on the available space
+2. Adaptive - Make content targeted for specific screen sizes so it'll render exactly like that, regardless of the available space.
+
+This RFC will focus on what approach and strategies we shall take in our Design System.
+
+# Basic Example
+
+### Responsive Modal
```

Comment:
Instead of specifying "Modal" let's just call it an example for "Responsive Design". Looking at "Responsive Modal" I was expecting an "Adaptive Modal" after this.
```suggestion
### Responsive Design
```